### ISSUE TITLE
sometimes p_energy is NaN

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#124]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/124) by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

When p_energy contains NaN values, then histogram doesn't work, neither does `ds.p_energy[:]<elo`. It either shouldn't ever be NaN, or we should have some workarounds. I've been doing `ds.p_energy[np.isnan(ds.p_energy)]=0` just to avoid problems.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Whose fault is this? I'm guessing it's the fault of the calibration object. Now that Young has introduced analytic derivatives into the object, I cannot understand it at all.

Nevertheless, I will see if I can somehow introduce the 1-liner you just gave me into `EnergyCalibration.__call__` or wherever it's appropriate.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

Current workaround:
```
def ds_sanitize_energies(ds, elo=0,emax=2e4):
    ds.p_energy[np.isnan(ds.p_energy[:])]=-1
    bad = np.logical_and(ds.p_energy[:]<elo, ds.p_energy[:]>emax)
    ds.cuts.cut("energy",bad)
def data_sanitize_energies(data, elo=0, emax=2e4):
    for ds in data:
        ds_sanitize_energies(ds, elo, emax)
```

I looked at this a bit more. In one dataset all the traces with nan energy have negative p_filt_value, and I plotted a few of the traces here:

[cid:image001.png@01D32309.716A0E10]

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

EnergyCalibration object cannot return NaN or inf.

Fix the ph2energy method to test for both outcomes and replace either 
with 0. Also make a test to be sure that neither 0 nor negative PHs
will produce NaN or inf energies. Fixes #124.

However, I should note that I never actually reproduced this bug.
